policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -39,7 +40,7 @@ struct NamespaceIndex {
 struct Namespace {
     service_routes: HashMap<ServicePort, ServiceRoutes>,
     namespace: Arc<String>,
-    services: HashMap<String, ServiceInfo>,
+    service_info: Arc<RwLock<HashMap<ServiceRef, ServiceInfo>>>,
```

Comment:
I don't totally get why this needs to be stored in a field on the namespace as well as on the index itself. The `Index` already lives inside a `RwLock`, and the `Index` owns all the namespaces, so the `service_info` map will only ever be mutated when the `Index` itself is being mutated, right? 

All the methods on `Namespace` are only called by the `Index`, so can't we remove the `Arc<RwLock<...>>` from around the `service_info` map and have the methods on `Namespace` that need access to it take the map as a reference that's passed in by the `Index` when calling the various `Namespace` methods?